pkg/clients/zone/zone.go

```diff
@@ -0,0 +1,101 @@
+package zone
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	"github.com/aws/aws-sdk-go-v2/service/route53/route53iface"
+	"github.com/google/uuid"
+
+	"github.com/crossplane/provider-aws/apis/network/v1alpha3"
+)
+
+// Client defines Route53 Client operations
+type Client interface {
```

Comment:
I don't think we need a level of indirection here. We usually implement only a fraction of the interface that AWS already provides and then in controllers we use methods to produce input for those functions. See RDS client interface here https://github.com/crossplane/provider-aws/blob/118e08d/pkg/clients/rds/rds.go#L39

Reply:
This has been changed accordingly, as suggested.